pilot/pkg/xds/discovery.go

```diff
@@ -467,6 +467,9 @@ func doSendPushes(stopCh <-chan struct{}, semaphore chan struct{}, queue *PushQu
 func (s *DiscoveryServer) initPushContext(req *model.PushRequest, oldPushContext *model.PushContext, version string) (*model.PushContext, error) {
 	push := model.NewPushContext()
 	push.PushVersion = version
+	if oldPushContext != nil {
+		oldPushContext.Cleanup()
```

Comment:
it might be good to comment that this is knowingly too early. Possible also on cleanup function, so someone doesn't decide to put something that will explode if that happens in the future